app/controllers/week_controller.rb

```diff
@@ -0,0 +1,18 @@
+class WeekController < ApplicationController
+  protect_from_forgery with: :exception
+
+  include AssignmentGenerationHelper
+
+  def new
+    last_week = Week.order(:starts_at).last #the last week added to the db
+
+    new_week = last_week.starts_at.next_week.beginning_of_week
+
+    week = Week.create(starts_at: new_week)
+
+    generate_new_week_assignments(week)
```

Comment:
I like the approach here to separate complicated business logic from the controller. A better tool for this than a helper method is an object. Please create an object that duplicates the behaviour you're getting with the helper but uses this line of code: `GenerateNewWeekAssignments.new(week: week).call`.

An object is nicer because we can create as many methods as we need to accomplish the task without adding those methods to the controller. Helpers are also added to all views and we don't want views to be generating database records.